plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -272,7 +277,30 @@ private TupleDomain<HiveColumnHandle> buildTupleDomainColumnHandle(EntryType ent
                         type)),
                 ColumnType.REGULAR,
                 column.getComment());
-        return TupleDomain.withColumnDomains(ImmutableMap.of(handle, Domain.notNull(handle.getType())));
+
+        ImmutableMap.Builder<HiveColumnHandle, Domain> domains = ImmutableMap.<HiveColumnHandle, Domain>builder()
+                .put(handle, Domain.notNull(handle.getType()));
+        if (entryType == ADD) {
+            partitionConstraint.getDomains().ifPresent(domain -> domain.forEach((key, value) -> domains.put(toPartitionValuesParsedField(handle, key), value)));
```

Comment:
pass `column` instead of `handle` into `toPartitionValuesParsedField`
(`handle` corresponds to "add entry's path field" while `column` corresponds to "add entry")